python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py

```diff
@@ -14,6 +14,8 @@
 from dagster._core.events import PIPELINE_RUN_STATUS_TO_EVENT_TYPE, DagsterEvent
 from dagster._core.instance import DagsterInstance
 from dagster._core.storage.pipeline_run import DagsterRun, DagsterRunStatus, PipelineRun, RunsFilter
+
+# from dagster._core.host_representation import JobSelector, RepositorySelector
```

Comment:
I was thinking it would make sense to use the existing RepositorySelector and make a new JobSelector, but the classes in host_representation aren't made public and attempting to import them here causes a circular import

Reply:
You can tackle this with the same mypy workaround, by using strings instead of type objects, and then importing the class in a `TYPE_CHECKING` conditional block (as opposed to on import).